### DISCUSSION TITLE
why the index page keep generate static html on each request

### DISCUSSION THREAD

**Discussion by yjisme** [vercel/next.js#10962] [Help] [answered]
add `page/index.js` to empty project, and includes these code:

```
export default ({ number }) => {
  console.log("index render");
  return (
    <div>
      <h1>Index : {number}</h1>
    </div>
  );
};

export async function getStaticProps() {
  console.log("index getStaticProps");
  return {
    props: {
      number: 1
    }
  };
}
```

then I use `next build` command, it's print correctly in console.

`next start` also work well

but when I request `http://localhost:3000/` in browser, the console log repeat every request except the first request.

i check `.next` dir, the `index.html` generated correctly, but why the request trigger it generate again?

but when I copy the code to any other page, there is no problem, no console words out

did i miss something?

**Comment by Timer** [comment, marked as answer]
Hey! Thanks for finding this bug. 

This is not the expected behavior and only is affecting `pages/index.js`—if you rename the page to `pages/named.js` (or similar) you should see it not re-render.

We've fixed this behavior in the latest Next.js canary: `next@^9.3.1-canary.1`.

Can you please give that a try?

**Reply by yjisme** [reply]
yeah! it's fixed!
Automatic Static Optimization is powerfull, you guys doing very well
